### PR TITLE
fix: dialog modal component property change

### PR DESCRIPTION
**Pull request by irenerobb72**
### Overview

Changing the Dialog Modal to have a href property and adds the danger variant

### PR Checklist

- [ ] Related to designs:
- [ ] Related to JIRA ticket: https://codecademy.atlassian.net/browse/LX-5108 
- [ ] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change

### CODE REVIEW

File: packages/gamut/src/Dialog/index.tsx [hunk 1]
```diff
@@ -11,6 +11,10 @@ import { Text } from '../Typography';
 
 const ShroudedOverlay = styled(Overlay)(system.css({ bg: 'shadow-opaque' }));
 
+type DialogButtonProps = Pick<
+  ComponentProps<typeof FillButton>,
+  'href' | 'onClick' | 'children'
+>;
```
Comment (codecaaron) [inline packages/gamut/src/Dialog/index.tsx:17]: Nit - Room to breathe
```suggestion
type DialogButtonProps = Pick<
  ComponentProps<typeof FillButton>,
  'href' | 'onClick' | 'children'
>;

```

**Review by codecaaron — APPROVED** [review]
Small formatting thing but otherwise this looks great!

**Review by dreamwasp — APPROVED** [review]
🙌

File: packages/gamut/src/Dialog/index.tsx [hunk 2]
```diff
@@ -11,6 +11,11 @@ import { Text } from '../Typography';
 
 const ShroudedOverlay = styled(Overlay)(system.css({ bg: 'shadow-opaque' }));
 
+interface DialogButtonProps {
+ children: FillButton['children'];
+ onClick?: FillButton['onClick'];
+ href?: FillButton['href'];
```
Comment (codecaaron) [inline packages/gamut/src/Dialog/index.tsx:17]:
```suggestion
 children: typeof FillButton['children'];
 onClick?: typeof FillButton['onClick'];
 href?: typeof FillButton['href'];
```

**Pull request closed without merging**